tensorflow/go/attrs.go

```diff
@@ -0,0 +1,258 @@
+/*
+Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tensorflow
+
+// #include <stdlib.h>
+// #include "tensorflow/c/c_api.h"
+//
+// TF_AttrType attrMetadataType(TF_AttrMetadata meta) {
+//   return meta.type;
+// }
+import "C"
+import (
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+type AttrType C.TF_AttrType
```

Comment:
For now, we've maintained a 1:1 correspondence between a Go type and an attribute type, which is why we don't need to export an "attribute type" concept when setting attributes (https://github.com/tensorflow/tensorflow/blob/5fa7b03a255d3c0d05aa48e7604a94185ef6b9e2/tensorflow/go/graph.go#L235).

I'd be tempted to maintain that here. Can we do away with all the new types here and instead have a single:

```go
func (op *Operation) Attr(name string) (interface{}, error)
```
?

Why bother with `IsList`, `ListSize`, `TotalSize` etc.?

Reply:
Done.